packages/api/src/platforms/vtex/resolvers/validateCart.ts

```diff
@@ -246,17 +250,42 @@ const getOrderForm = async (
 export const validateCart = async (
   _: unknown,
   { cart: { order }, session }: MutationValidateCartArgs,
-  ctx: Context,
+  ctx: Context
 ) => {
   const { enableOrderFormSync } = ctx.storage.flags
   const { orderNumber, acceptedOffer, shouldSplitItem } = order
   const {
     clients: { commerce },
     loaders: { skuLoader },
+    headers,
   } = ctx
 
   // Step1: Get OrderForm from VTEX Commerce
   const orderForm = await getOrderForm(orderNumber, session, ctx)
+  const cookieSession = await getCookie('vtex_session', headers.cookie)
+
+  if (cookieSession) {
```

Comment:
I chose to remove the enable OrderForm Sync flag from validity, but if you think it makes sense it can be included at this point.